src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.NativeAot.cs

```diff
@@ -361,6 +361,26 @@ public static void PrepareDelegate(Delegate d)
 
             return RuntimeImports.RhNewObject(mt);
         }
+
+        public static unsafe object Box(ref byte target, RuntimeTypeHandle type)
+        {
+            if (type.IsNull)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.type);
+
+            // Compatibility with CoreCLR, throw on a null reference to the unboxed data.
+            if (Unsafe.IsNullRef(ref target))
+                throw new NullReferenceException();
+
+            MethodTable* eeType = type.ToMethodTable();
+
+            if (eeType->IsByRefLike)
+                throw new NotSupportedException(SR.NotSupported_ByRefLike);
+
+            if (eeType->IsPointer || eeType->IsFunctionPointer)
+                throw new NotSupportedException(SR.NotSupported_BoxedPointer);
+
+            return RuntimeImports.RhBoxAny(ref target, eeType);
```

Comment:
It may be nice to inline the non-value type case here to make it fast and match CoreCLR (RuntimeImports are not inlineable).